### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by azubieta**
Datails in the commit names.

### CODE REVIEW

**Comment by limpkin** [comment]
@raoulh this is a WIP, do not merge.

**Comment by limpkin** [comment]
@raoulh : ready to be merged!

File: Makefile [hunk 1]
```diff
@@ -87,6 +87,10 @@ deb_changelog: git_setup
 deb_package:
 	$(DOCKER_EXEC) "cp -f README.md debian/README"
 	$(DOCKER_EXEC) "dpkg-buildpackage -b -us -uc && mkdir -p build-linux/deb && cp ../*.deb build-linux/deb"
+	
+# Custom upload
+custom_upload:
+	-. scripts/ci/funcs.sh && upload_file build-linux/deb/$(DEB_NAME) $(sha256sum build-linux/deb/$(DEB_NAME) | cut -d' ' -f1) "linux"
```
Comment (raoulh) [inline Makefile]: Why is this in double here? Wong merge?

File: src/MainWindow.cpp [hunk 2]
```diff
@@ -105,6 +105,9 @@ MainWindow::MainWindow(WSClient *client, QWidget *parent) :
     connect(wsClient, &WSClient::statusChanged, [this]()
     {
         this->enableKnockSettings(wsClient->get_status() == Common::NoCardInserted);
+        if (wsClient->get_status() == Common::UnkownSmartcad) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: Please follow existing code style, { on a newline

File: src/MainWindow.cpp [hunk 3]
```diff
@@ -769,6 +761,18 @@ void MainWindow::wantSaveCredentialManagement()
     ui->progressBarWait->hide();
 
     connect(wsClient, SIGNAL(progressChanged(int,int)), this, SLOT(loadingProgress(int,int)));
+
+    auto conn = std::make_shared<QMetaObject::Connection>();
+    *conn = connect(wsClient, &WSClient::credentialsUpdated, [this, conn](const QString & , const QString &, const QString &, bool success)
+    {
+        disconnect(*conn);
+        if (!success) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: } on newline

File: src/MainWindow.cpp [hunk 4]
```diff
@@ -1022,6 +1032,48 @@ void MainWindow::enableCredentialsManagement(bool enable)
 
     if (!enable)
         updatePage();
+
+    updateTabButtons();
+}
+
+void MainWindow::updateTabButtons()
+{
+    auto setEnabledToAllTabButtons = [=](bool enabled) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: } on newline

File: src/MainWindow.cpp [hunk 5]
```diff
@@ -1022,6 +1032,48 @@ void MainWindow::enableCredentialsManagement(bool enable)
...
+        for (QObject * object: ui->widgetHeader->children()) {
+            if (typeid(*object) ==  typeid(QPushButton)) {
+                QAbstractButton *tabButton = (QAbstractButton *) object;
+                tabButton->setEnabled(enabled);
+            }
+        }
+    };
+
+    if (ui->stackedWidget->currentWidget() == ui->pageWaiting) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: } on newline

File: src/MainWindow.cpp [hunk 6]
```diff
@@ -1022,6 +1032,48 @@ void MainWindow::enableCredentialsManagement(bool enable)
...
+        setEnabledToAllTabButtons(false);
+    } else
+        setEnabledToAllTabButtons(true);
+
+    // Enable or Disable tabs according to the device status
+    if (wsClient->get_status() == Common::UnkownSmartcad) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: } on newline

File: src/MainWindow.cpp [hunk 7]
```diff
@@ -1022,6 +1032,48 @@ void MainWindow::enableCredentialsManagement(bool enable)
...
+        setEnabledToAllTabButtons(true);
+
+        ui->pushButtonCred->setEnabled(false);
+        ui->pushButtonFiles->setEnabled(false);
+        ui->pushButtonSSH->setEnabled(false);
+
+        return;
+    }
+
+    if (wsClient->get_memMgmtMode()) {
```
Comment (raoulh) [inline src/MainWindow.cpp]: } on newline

File: src/WSServerCon.cpp [hunk 8]
```diff
@@ -872,6 +872,12 @@ void WSServerCon::sendFilesCache()
         return;
     }
 
+    auto deviceStatus = mpdevice->get_status();
+    if (deviceStatus != Common::Unlocked) {
```
Comment (raoulh) [inline src/WSServerCon.cpp]: } on newline

**Comment by limpkin** [comment]
Will correct everything

**Pull request merged**